frontend/src/state/documents.ts

```diff
@@ -34,6 +34,41 @@ export function createDocumentsState(editor: EditorState) {
 		download(triggerFileDownload.name, triggerFileDownload.document);
 	});
 
+	editor.dispatcher.subscribeJsMessage(TriggerRasterDownload, (triggerRasterDownload) => {
+		// A canvas to render our svg to in order to get a raster image
+		// https://stackoverflow.com/questions/3975499/convert-svg-to-image-jpeg-png-etc-in-the-browser
+		const canvas = document.createElement("canvas");
+		canvas.width = triggerRasterDownload.size.x;
+		canvas.height = triggerRasterDownload.size.y;
+		const ctx = canvas.getContext("2d");
+		if (!ctx) return;
+
+		// Fill the canvas with white if jpeg (does not support transparency and defaults to black)
+		if (triggerRasterDownload.mime.endsWith("jpeg")) {
+			ctx.fillStyle = "rgb(255,255,255)";
```

Comment:
Nitpick: may as well use the shorter/clearer shorthand of `"white"`. Obviously this unimportant but I thought I'd mention it.